servers/dsub/jobs/controllers/aggregation_controller.py

```diff
@@ -24,12 +25,79 @@ def get_job_aggregations(timeFrame, projectId=None):
     Returns:
         AggregationResponse: Response contains aggregation of jobs
     """
+    query = QueryJobsRequest.from_dict({
+        'extensions': {
+            'project_id': projectId
+        }
+    })
+
+    query.start = time_convert.time_frame_to_start_time(timeFrame)
+    query.extensions.project_id = projectId
+
+    provider = providers.get_provider(generator.provider_type(), projectId,
+                                      generator.auth_token())
+    job_generator = generator.generate_jobs(provider, query)
+
+    total_count = 0
+
+    summary_counts = {}
+
+    # print(type(job_statuses.ApiStatus))
+    # for status in job_statuses.ApiStatus:
+    #     print(status)
+
+    for job in job_generator:
+        total_count += 1
+        _summary_count(job, summary_counts)
+
+    print("total: ", total_count, ", details: ", summary_counts)
+
+    return _get_stub_data(_to_summary_counts(summary_counts))
+
+
+def _summary_count(job, summary_counts):
+    if job.status == job_statuses.ApiStatus.SUBMITTED:
+        summary_counts[
+            job_statuses.ApiStatus.SUBMITTED] = summary_counts.setdefault(
+                job_statuses.ApiStatus.SUBMITTED, 0) + 1
+    elif job.status == job_statuses.ApiStatus.RUNNING:
+        summary_counts[
+            job_statuses.ApiStatus.RUNNING] = summary_counts.setdefault(
+                job_statuses.ApiStatus.RUNNING, 0) + 1
+    elif job.status == job_statuses.ApiStatus.ABORTING:
+        summary_counts[
+            job_statuses.ApiStatus.ABORTING] = summary_counts.setdefault(
+                job_statuses.ApiStatus.ABORTING, 0) + 1
+    elif job.status == job_statuses.ApiStatus.ABORTED:
+        summary_counts[
+            job_statuses.ApiStatus.ABORTED] = summary_counts.setdefault(
+                job_statuses.ApiStatus.ABORTED, 0) + 1
+    elif job.status == job_statuses.ApiStatus.SUCCEEDED:
+        summary_counts[
+            job_statuses.ApiStatus.SUCCEEDED] = summary_counts.setdefault(
+                job_statuses.ApiStatus.SUCCEEDED, 0) + 1
+    elif job.status == job_statuses.ApiStatus.FAILED:
+        summary_counts[
+            job_statuses.ApiStatus.FAILED] = summary_counts.setdefault(
+                job_statuses.ApiStatus.FAILED, 0) + 1
+
+
+def _to_summary_counts(summary_counts):
+    counts = []
+
+    for status, count in summary_counts.items():
```

Comment:
The usual way to do this in Python is to use a list comprehension (simple enough that I would probably inline it)  `[x for x in y]` (I forget the exact iterator you'll need for a dict).

Reply:
Thanks!